openai/openai.go

```diff
@@ -54,17 +60,17 @@ type ResponseFormat struct {
 }
 
 type ChatCompletionRequest struct {
-	Model            string          `json:"model"`
-	Messages         []Message       `json:"messages"`
-	Stream           bool            `json:"stream"`
-	MaxTokens        *int            `json:"max_tokens"`
-	Seed             *int            `json:"seed"`
-	Stop             any             `json:"stop"`
-	Temperature      *float64        `json:"temperature"`
-	FrequencyPenalty *float64        `json:"frequency_penalty"`
-	PresencePenalty  *float64        `json:"presence_penalty_penalty"`
-	TopP             *float64        `json:"top_p"`
-	ResponseFormat   *ResponseFormat `json:"response_format"`
+	Model            string         `json:"model"`
+	Messages         []Message      `json:"messages"`
+	Stream           bool           `json:"stream"`
+	MaxTokens        *int           `json:"max_tokens"`
+	Seed             *int           `json:"seed"`
+	Stop             any            `json:"stop"`
+	Temperature      *float32       `json:"temperature"`
```

Comment:
just to avoid any breakages, let's leave this out of the change for now since it affects chat completions